src/components/VariantSearchField.tsx

```diff
@@ -25,10 +25,157 @@ const backgroundColor: { [key in SearchType]: string } = {
   'nuc-mutation': 'rgba(33,162,162,0.29)',
 };
 
+interface NspMap {
+  [name: string]: number;
+}
+
+const nsps: NspMap = {
+  nsp1: 1,
+  nsp2: 181,
+  nsp3: 819,
+  nsp4: 2764,
+  nsp5: 3264,
+  nsp6: 3570,
+  nsp7: 3860,
+  nsp8: 3942,
+  nsp9: 4141,
+  nsp10: 4254,
+  nsp12: 4393,
+  nsp13: 5325,
+  nsp14: 5926,
+  nsp15: 6453,
+  nsp16: 6799,
+};
+
+function getEquivalent(value: string, notation: string) {
+  let orf1aorbcodon: number | null = null;
+  let nsp: string | null = null;
+  let nspCodon: number | null = null;
+  let combinedCodon: number | null = 3646;
+  let mutationArray = value.split(':');
+  if (mutationArray[0] === 'ORF1b' || mutationArray[0] === 'ORF1a') {
+    let letterBefore = mutationArray[1].charAt(0);
+    let letterAfter = mutationArray[1].charAt(mutationArray[1].length - 1);
+    orf1aorbcodon = parseInt(mutationArray[1].slice(1, -1));
+    combinedCodon = orf1aorbcodon + (mutationArray[0] === 'ORF1b' ? 4401 : 0);
+    nsp = 'nsp1';
+    nspCodon = combinedCodon - nsps[nsp] + 1;
+    for (const [key, value] of Object.entries(nsps)) {
+      if (combinedCodon > value && combinedCodon - value < combinedCodon - nsps[nsp]) {
+        nsp = key;
+        nspCodon = combinedCodon - value + 1;
+      }
+    }
+    if (notation === 'nsp') {
+      return `${nsp}:${letterBefore}${nspCodon}${letterAfter}`;
+    } else if (notation === 'orf1ab') {
+      return `ORF1ab:${letterBefore}:${combinedCodon}${letterAfter}`;
+    } else {
+      return value;
+    }
+  }
+}
+
+const translateMutation = (oldValue: string) => {
```

Comment:
This function shares some code with `getEquivalent()`. Is it possible to reduce code redundancy?